libs/api/ebpf_api.cpp

```diff
@@ -1397,6 +1397,13 @@ _clean_up_ebpf_object(_In_opt_ ebpf_object_t* object) noexcept
         clean_up_ebpf_programs(object->programs);
         clean_up_ebpf_maps(object->maps);
 
+        if (object->native_module_handle != ebpf_handle_invalid) {
+            ebpf_assert(
+                object->execution_type == EBPF_EXECUTION_NATIVE || object->execution_type == EBPF_EXECUTION_ANY);
```

Comment:
when would you store a native module handle but store execution_type of ANY?  Seems like we should always store the real one (NATIVE)

Reply:
I agree. Just that `ebpf_object_set_execution_type()` allowed `EBPF_EXECUTION_ANY` to be set for a native module, so I included that option also in the assert.

Will create a separate PR to update API `ebpf_object_set_execution_type()` and update the assert here.

Reply:
On second thought, I think EBPF_EXECUTION_ANY should be allowed so that a user can set this execution type, without worrying about the actual file type being used (`.o` vs. `.sys`). By setting EBPF_EXECUTION_ANY, a user is agreeing to whatever is the default execution type based on the file type and a system wide policy, if any.

Reply:
Sure, but when setting EBPF_EXECUTION_ANY on a native file, it should store EBPF_EXECUTION_NATIVE.  In other words, we should never store EBPF_EXECUTION_ANY in my view, the default should be stored.  So if you call ebpf_object_set_execution_type(EBPF_EXECUTION_ANY) then ebpf_object_get_execution_type() you should be able to see what the default is based on what is returned.

Reply:
Created PR#1561 to fix the behavior.

Reply:
Fixed the assert here.